internal/metrics/metrics.go

```diff
@@ -203,19 +204,29 @@ func (m *Metrics) SetIngressRouteMetric(metrics IngressRouteMetric) {
 type Service struct {
 	httpsvc.Service
 	*prometheus.Registry
+	Client *kubernetes.Clientset
 }
 
 // Start fulfills the g.Start contract.
 // When stop is closed the http server will shutdown.
 func (svc *Service) Start(stop <-chan struct{}) error {
-	registerHealthCheck(&svc.ServeMux)
+
+	registerHealthCheck(&svc.ServeMux, svc.Client)
 	registerMetrics(&svc.ServeMux, svc.Registry)
 
 	return svc.Service.Start(stop)
 }
 
-func registerHealthCheck(mux *http.ServeMux) {
+func registerHealthCheck(mux *http.ServeMux, client *kubernetes.Clientset) {
 	mux.HandleFunc("/health", func(w http.ResponseWriter, r *http.Request) {
+		// Try and lookup Kubernetes server version as a quick and dirty check
+		_, err := client.ServerVersion()
+		if err != nil {
+			w.WriteHeader(http.StatusServiceUnavailable)
```

Comment:
Could you also please include the error message. we get from client.ServerVersion()?

There is a helper in the http package for this
```go
msg := fmt.Sprintf("Failed Kubernetes Check: %v", err)
http.Error(w, msg, http.StatusServiceUnavailable)
return
```